### ISSUE TITLE
Add support for pinning the infra container image

### ISSUE THREAD

**Issue by afbjorklund** [Mirantis/cri-dockerd#39]
KEP 2040: Kubelet CRI support

https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2040-kubelet-cri#pinned-images

* https://github.com/kubernetes/kubernetes/issues/106893

Requires CRI API 1.23

**Comment by afbjorklund** [comment]
Currently cri-dockerd is using pause version 3.1, while e.g. containerd is using pause version 3.6

**Comment by hakman** [comment]
@evol262 Maybe this should not be closed yet?

**Comment by afbjorklund** [comment]
As far as I know, this is what the `--pod-infra-container-image` option to cri-dockerd is (still) doing.

**Comment by afbjorklund** [comment]
The current kubelet flag will be removed in 1.30

**Comment by afbjorklund** [comment]
Good thing that the "pause" image is normally in use, now when the cleaning has started:

```
REPOSITORY                                TAG               IMAGE ID       CREATED         SIZE
registry.k8s.io/pause                     3.9               e6f181688397   13 months ago   744kB
```

"Disk usage on image filesystem is over the high threshold, trying to free bytes down to the low threshold":

```
Nov 19 14:22:45 minikube kubelet[5545]: I1119 14:22:45.466491    5545 image_gc_manager.go:458] "Removing image to free bytes" imageID="sha256:e6f1816883972d4be47bd48879a08919b96afcd344132622e4d444987919323c" size=743952 runtimeHandler=""
Nov 19 14:22:45 minikube kubelet[5545]: E1119 14:22:45.468675    5545 remote_image.go:212] "RemoveImage from image service failed" err="rpc error: code = Unknown desc = Error response from daemon: conflict: unable to remove repository reference \"registry.k8s.io/pause:3.9\" (must force) - container 6fae6463b255 is using its referenced image e6f181688397" image="sha256:e6f1816883972d4be47bd48879a08919b96afcd344132622e4d444987919323c"
Nov 19 14:22:45 minikube kubelet[5545]: E1119 14:22:45.468713    5545 kuberuntime_image.go:138] "Failed to remove image" err="rpc error: code = Unknown desc = Error response from daemon: conflict: unable to remove repository reference \"registry.k8s.io/pause:3.9\" (must force) - container 6fae6463b255 is using its referenced image e6f181688397" image="sha256:e6f1816883972d4be47bd48879a08919b96afcd344132622e4d444987919323c"
Nov 19 14:22:45 minikube kubelet[5545]: E1119 14:22:45.468733    5545 kubelet.go:1442] "Image garbage collection failed multiple times in a row" err="wanted to free 47879980646 bytes, but freed 0 bytes space with errors in image deletion: rpc error: code = Unknown desc = Error response from daemon: conflict: unable to remove repository reference \"registry.k8s.io/pause:3.9\" (must force) - container 6fae6463b255 is using its referenced image e6f181688397"
```

`Server Version: v1.29.0-alpha.3`

**Comment by brandond** [comment]
Is anyone at Mirantis actively working on this, or is this going to wait for someone in the community to push it across the line?

**Comment by afbjorklund** [comment]
Now included in `v0.3.10`
